src/js/models/post.js

```diff
@@ -217,10 +207,27 @@ export default class Post {
     const post = this.builder.createPost();
     const { builder } = this;
 
-    this.walkPostSections(range, section => {
+    let sectionParent = post,
+        listParent = null;
+    this.walkLeafSections(range, section => {
```

Comment:
haha, for a different time but I wonder if this can be structured as an abstract renderer :-)